Fix unit test TestCertsTLS

CaoShuFeng opened the pull request:
fixes #47304 
**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```
NONE
```

kargakis commented:
@k8s-bot ok to test

madhusudancs commented on federation/pkg/kubefed/init/init_test.go:
```diff
@@ -1444,7 +1444,8 @@ func tlsHandshake(t *testing.T, sCfg, cCfg *tls.Config) error {
 	}()
 
-	c, err := tls.Dial("tcp", s.Addr().String(), cCfg)
+	addr := strings.Split(s.Addr().String(), ":")
```
I looked at the issue filed. It looks like you somehow have network protocol default to IPv6 in your environment. Shouldn't you make that default to IPv4 when running these tests instead of this hacky solution in the code base?

Even if you were to make a change here, the right way to do this is to force the address to IPv4 by passing `tcp4` instead of `tcp` to `tls.Listen()` call above.

CaoShuFeng replied:
Thanks for your guidance.
Your are right. When I enable ipv6, this test passes in my environment.

I passed tcp4 to tls.Listen() for the guys who disables ipv4.

CaoShuFeng committed:
Fix unit test TestCertsTLS when IPV6 is disabled

tls.Listen returns an ipv6 address, this will make the test fail when
ipv6 is disabled.

madhusudancs commented on federation/pkg/kubefed/init/init_test.go:
```diff
@@ -1418,7 +1418,7 @@ func tlsHandshake(t *testing.T, sCfg, cCfg *tls.Config) error {
 	// instead. One could have as well used a Unix Domain Socket, but TCP is
 	// more portable.
-	s, err := tls.Listen("tcp", "", sCfg)
+	s, err := tls.Listen("tcp4", "", sCfg) // specify tcp4 for the guys who disable ipv6
```
`who have enabled IPv6` you mean in your comment?

Also, can you put that comment above this line?

madhusudancs reviewed (commented):
> I passed tcp4 to tls.Listen() for the guys who disables ipv6.

I am not entirely sure if that's what is happening? If someone disabled IPv6, they should always get IPv4 addresses and everything should already work fine with `tcp` instead of `tcp6` right?

sttts commented:
Related: https://github.com/golang/go/issues/18806
